server/controllers/medical/patients/checkin.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @module medical/patients/checkin
+ * 
+ * @description
+ * This controller is responsible for the Patient Check In feature, this allows a hospital to track when returning
+ * patients visit the hospital.
+ *
+ * It is responsible for reading and writing to the `patient_visit` database table as well as responding to HTTP
+ * requests.
+ *
+ * @requires  node-uuid
+ * @requires  lib/db
+ * @requires  lib/topic
+ * @requires  lib/errors/BadRequest
+ */
+'use strict'
+
+const uuid  = require('node-uuid');
+
+const db    = require('../../../lib/db');
+const topic = require('../../../lib/topic');
+
+const BadRequest = require('../../../lib/errors/BadRequest');
+
+exports.list = list;
+exports.create = create;
+
+/**
+ * @method list
+ *
+ * @description
+ * List all records of the patients visit given a specified identifier.
+ *
+ * GET /patients/:uuid/visits
+ */
+function list(req, res, next) {
+  const patientUuid = req.params.uuid;
+  let limitQuery = '';
+
+  if (req.query.limit) {
+    
+    // validate query string is valid - do not template in anything sent from the client
+    if (isNaN(Number(req.query.limit))) { 
```

Comment:
A slightly clearer and more secure way (although, a dedicated attacker could likely find a way around it) is:
```js
const limit = Number(req.query.limit);

if (isNaN(limit)) {
  throw new BadRequest('...');
}

limitQuery = `LIMIT ${limit}`;
```
This won't block the PR, since the attack would be quite contrived.